### PR TITLE
[JENKINS-45157] [WIP] PCT should be able to exercise declarative pipeline plugins

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
[JENKINS-45157](https://issues.jenkins-ci.org/browse/JENKINS-45157)

Same approach as used in #34 

@reviewbybees specially @varyvol and @kwhetstone

### CODE REVIEW

**Comment by kwhetstone** [comment]
:bee:  Let's wait to merge #34 and then we'll merge this one.  This looks fine from an implementation perspective.

File: plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/TransformPom.java [hunk 1]
```diff
@@ -25,6 +25,7 @@ public boolean check(Map<String, Object> info) {
         PomData pomData = (PomData)info.get("pomData");        
         MavenCoordinates parent = pomData.parent;
         MavenCoordinates coreCoordinates = (MavenCoordinates)info.get("coreCoordinates");
+        boolean isDeclarartivePipeline = parent.matches("org.jenkinsci.plugins", "pipeline-model-parent");
```
Comment (jglick) [inline plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/TransformPom.java]: Typo.

**Comment by raul-arabaolaza** [comment]
*@jglick *In fact I am not trying more profound fixes or refactorings
because AFAIK you are working in a new approach.

2017-06-28 21:41 GMT+02:00 Jesse Glick <notifications@github.com>:

**Comment by jglick** [comment]
Right, my _hope_ is that we will able to run something along the lines of

```sh
git clone https://github.com/jenkinsci/pipeline-model-definition-plugin
mvn -f pipeline-model-definition-plugin -DoverrideVersions=org.jenkins-ci.main:jenkins-core:2.60.1,org.jenkins-ci.main:jenkins-war:2.60.1,org.jenkins-ci.plugins:branch-api:2.0,… -DuseUpperBounds=true clean install
```

(to run all tests in all modules in that reactor using updated dependencies) but the concept is not yet proven.

**Comment by raul-arabaolaza** [comment]
@jglick That sounds cool given that is able to deal with different types of dependencies (optional, transitive) , scopes and classifiers. Also the list of dependencies can be large and generate it manually can be very error prone. So I would suggest implementing a mechanism similar to `GIT_ASKPASS` to get the override list, this way we can implement a way to take the last versions deployed in a maven repo or the versions deployed into a running jenkins instance

**Comment by jglick** [comment]
Why closed?

**Comment by raul-arabaolaza** [comment]
@jglick Superseeded by #37

**Pull request closed without merging**